From 446e61ba07ec949544594729cdd2dff79c8e4cdc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 15 Jul 2014 09:52:17 -0400
Subject: [PATCH 22/32] timesyncd: only listen to clock changes when connected

This reverts previous commit and applies a different fix.

manager_clock_watch() callback calls manager_send_request() to kick
off a resync. We can only do that when we're actually connected to
something. It is not useful to setup the callback from manager_new().

Now the callback will be dropped in manager_connect() and requested
in manager_begin().

https://bugs.freedesktop.org/show_bug.cgi?id=80932
---
 src/timesync/timesyncd.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/timesync/timesyncd.c b/src/timesync/timesyncd.c
index 9bc773c..19af9f9 100644
--- a/src/timesync/timesyncd.c
+++ b/src/timesync/timesyncd.c
@@ -331,9 +331,6 @@ static int manager_clock_watch(sd_event_source *source, int fd, uint32_t revents
         /* rearm timer */
         manager_clock_watch_setup(m);
 
-        if (!m->current_server_address)
-                return 0;
-
         /* skip our own jumps */
         if (m->jumped) {
                 m->jumped = false;
@@ -1047,10 +1044,6 @@ static int manager_new(Manager **ret) {
         if (r < 0)
                 return r;
 
-        r = manager_clock_watch_setup(m);
-        if (r < 0)
-                return r;
-
         *ret = m;
         m = NULL;
 
-- 
1.8.5.5

